api-server/scanner-lib/src/blockchain_state/mod.rs

```diff
@@ -546,40 +544,42 @@ async fn update_tables_from_consensus_data<T: ApiServerStorageWrite>(
             distribute_pos_reward(&mut adapter, block.get_id(), pool_id, total_reward)
                 .expect("no error");
 
-            for (delegation_id, rewards, updated_delegation) in adapter.rewards_per_delegation() {
+            for (delegation_id, updated_delegation) in adapter.delegations_iter() {
                 db_tx
-                    .set_delegation_at_height(delegation_id, &updated_delegation, block_height)
+                    .set_delegation_at_height(*delegation_id, updated_delegation, block_height)
                     .await?;
+
                 let address = Address::<Destination>::new(
                     &chain_config,
                     updated_delegation.spend_destination(),
                 )
                 .expect("Unable to encode address");
-                increase_address_amount(
-                    db_tx,
-                    &address,
-                    &rewards,
-                    CoinOrTokenId::Coin,
-                    block_height,
-                )
-                .await;
+                db_tx
+                    .set_address_balance_at_height(
+                        address.get(),
+                        *updated_delegation.balance(),
+                        CoinOrTokenId::Coin,
+                        block_height,
+                    )
+                    .await
+                    .expect("Unable to update balance");
```

Comment:
I think this is a mistake, you can't set the address balance to only the delegation balance, in the case that the same address has been reused for other things in the past.

Reply:
oh right, so the address is not unique and user can reuse it for multiple delegation? Interesting, in that case I indeed misunderstood the logic.

Reply:
yep, they can reuse the address for anything, normally it shouldn't happen if they are using the wallet but can't be sure what the user will do

Reply:
reverted